homeassistant/components/rabbitair/entity.py

```diff
@@ -0,0 +1,75 @@
+"""A base class for Rabbit Air entities."""
+from __future__ import annotations
+
+import asyncio
+import logging
+from typing import Any
+
+from rabbitair import Client, Model, State
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+MODELS = {
+    Model.A3: "A3",
+    Model.BioGS: "BioGS 2.0",
+    Model.MinusA2: "MinusA2",
+    None: None,
+}
+
+
+class RabbitAirBaseEntity(CoordinatorEntity[State]):
+    """Base class for Rabbit Air entity."""
+
+    def __init__(
+        self,
+        coordinator: DataUpdateCoordinator[State],
+        client: Client,
+        entry: ConfigEntry,
+    ) -> None:
+        """Initialize the entity."""
+        super().__init__(coordinator)
+        self._client = client
+        self._entry = entry
+        self._attr_unique_id = entry.unique_id
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, entry.unique_id)},  # type: ignore[arg-type]
```

Comment:
Why are we ignoring type here? Can we avoid doing that?

Reply:
This is because, according to the type information, `entry.unique_id` can be `None`, while `identifiers` can't. But we always fill in the `unique_id` when we create an entry, so it should not be `None` "by design". Perhaps we could use assert here? That would also remove the warning, and better show our intentions.

Reply:
Made a suggestion in the PR to avoid needing to do this (by storing the MAC in the entry data)